python-package/xgboost/compat.py

```diff
@@ -16,14 +16,18 @@
     # pylint: disable=invalid-name, redefined-builtin
     STRING_TYPES = (str,)
 
-
     def py_str(x):
         """convert c string back to python string"""
-        return x.decode('utf-8')
+        try:
+            s = x.decode('utf-8')
+        except UnicodeDecodeError as e:
+            print('Failed to decode error message, please file a bug report.')
```

Comment:
1. I am not sure it is very useful. I think the message should contain a bad string unless you expect it to happen often and the string is big
2. I would rather use logging library and pass and exception there as is. It usually produces a better output and one can configure it.
3. If you insist on using print can I suggest to use `stderr` instead of `stdout`?

Reply:
A friendly ping, what do you think about using a logging library?